### ISSUE TITLE
Request, adaptable resolution/fullscreen

### ISSUE THREAD

**Issue by Smeltie** [StevePaget/Pygame_Functions#16]
I thought it would be neat to add fullscreen and an adaptable resolution or atleast resolution settings.

the way i thought it could be done is by changing it ingame like clicking on a setting which changes the screensize value's.
and then make presets

and maybey a way to directly make the labels adapt to it? (but thats not necassary)

just fullscreen and adjustable resolution would be nice

**Comment by StevePaget** [comment]
At the moment, you can use a fullscreen mode:

screenSize(600,600, fullscreen=True)

But this still uses the window size setting, and simply magnifies it to fit the screen. I'm guessing you want it to detect the native resolution of the desktop and set the resolution automatically?

**Comment by Smeltie** [comment]
Exactly, i have been able to do something like it but it never worked well, it would always cut off pieces so you would miss half of the screen or the screen would be black with the background image only on a quarter of the screen, only when i make the background image my native resolution it starts to work well but then when you try it on a different computer it doesnt work.

**Comment by comport9** [comment]
So, in def screenSize I made this change:
```
os.environ['SDL_VIDEO_CENTERED'] = '1'
#os.environ['SDL_VIDEO_WINDOW_POS'] = "%d, %d" % ((monitorWidth - sizex) / 2, (monitorHeight - sizey) / 2)
if fullscreen:
	#screen = pygame.display.set_mode([sizex, sizey], pygame.FULLSCREEN)
	screen = pygame.display.set_mode([monitorWidth, monitorHeight+23])
```
monitorHeight+23 removed the top bar. Not elegant, but it works. However, it worked on MY system, other systems with different resolutions, I'm not sure about...

**Comment by xXLarryTFVWXx** [comment]
I don't know about with pygame_functions, but with vanilla pygame you can just do `pygame.display.set_mode((0,0))` for an automatic fullscreen

**Comment by Smeltie** [comment]
Oh wow, i didn’t expect a reply nearly 5 years later haha.
This school project ended ages ago, I’ve been working in IT for close to 3 years now.
Thanks for your time and dedication and I hope this project may serve people well for many years to come.
